daemon/endpoint.go

```diff
@@ -99,6 +103,48 @@ func (h *getEndpointID) Handle(params GetEndpointIDParams) middleware.Responder
 	}
 }
 
+type getEndpoints struct {
+	d *Daemon
+}
+
+// NewGetEndpoints handler returns a new getEndpoints populated with provided Daemon.
+func NewGetEndpointsHandler(d *Daemon) GetEndpointsHandler {
+	return &getEndpoints{d: d}
+}
+
+func (h *getEndpoints) Handle(params GetEndpointsParams) middleware.Responder {
+	log.Debugf("GET /endpoints request: %+v", params.Labels)
+
+	args := filters.NewArgs()
+	for _, v := range params.Labels {
+		args.Add("label", v)
+	}
+
+	containers, err := h.d.dockerClient.ContainerList(ctx.Background(), types.ContainerListOptions{
```

Comment:
This is not a good idea. This will potentially block this API call for a long time. We should fetch and store the docker labels with the endpoint when we receive the create event and then just look for local endpoints which match the labels.